QHyper/problems/community_detection.py

```diff
@@ -139,7 +140,7 @@ def _get_discrete_variable_representation(
         return sympy.symbols(" ".join([f"x{i}" for i in range(len(self.community))]))
 
     def _set_objective_function(self) -> None:
-        equation: dict[VARIABLES, float] = {}
+        equation = {}
```

Comment:
Currently, what can replace the VARIABLES type?

Reply:
We're no longer require variables in Problem. They are obtain from polynomial using `get_variables()`.
In this example equation looks like `dict[tuple[str, ...], float]`, which at the end should be converted to polynomial (because `objective_fuction` should be a polynomial)